### PR TITLE
Improvement/doc

### PR DESCRIPTION
**Pull request by TornaxO7**
just a general improvement

### CODE REVIEW

File: src/backend/mod.rs [hunk 1]
```diff
@@ -3,37 +3,59 @@
...
 pub enum Error {
     /// Represents that an error occured while trying to get a screenshot on X11.
     #[error(transparent)]
     X11(#[from] x11::Error),
 }
 
+/// An alias type for better code readability.
 pub type Pixel = u16;
 
+/// This enum contains specifique values depending on which backend.
```
Comment (eneoli) [inline src/backend/mod.rs:18]: Grammer

File: src/backend/mod.rs [hunk 2]
```diff
@@ -3,37 +3,59 @@
...
+    /// The height of the output.
     pub height: Pixel,
+
+    /// The x-value of the top-left corner of the output.
     pub x: i16,
+
+    /// The y-value of the top-left corner of the output.
     pub y: i16,
 
+    /// A general id of the output.
```
Comment (eneoli) [inline src/backend/mod.rs:43]: A Id? Isn't it THE Id of the output?

File: src/backend/mod.rs [hunk 3]
```diff
@@ -3,37 +3,59 @@
...
 
 /// The main function of this module.
 ///
-/// This function returns an rgb-image for each screen (or "monitor" in other
+/// # General description
+/// This function returns an image for each screen (or "monitor" in other
 /// words).
+///
+/// # Return value
+/// A tuple where the first value contains additional information about the image which is the second
```
Comment (eneoli) [inline src/backend/mod.rs:57]: Grammer

File: src/lib.rs [hunk 4]
```diff
@@ -1,7 +1,16 @@
+//! Welcome to the code-documentation of flakeshot!
```
Comment (eneoli) [inline src/lib.rs:1]: I don't think we can consider this file as a general entry point every reader will start with

**Review by eneoli — CHANGES_REQUESTED** [review]

File: src/backend/mod.rs (hunk 1 above)
Comment (TornaxO7) [reply src/backend/mod.rs:18]: fixed

File: src/lib.rs (hunk 4 above)
Comment (TornaxO7) [reply src/lib.rs:1]: Why not? It's the first page which pops up if you open the docs.

File: src/backend/mod.rs (hunk 3 above)
Comment (TornaxO7) [reply src/backend/mod.rs]: fixed?

File: src/lib.rs (hunk 4 above)
Comment (eneoli) [reply src/lib.rs:1]: Huh? Which page?

Comment (TornaxO7) [reply src/lib.rs:1]: just execute `cargo doc --open`

Comment (eneoli) [reply src/lib.rs:1]: Ha, didn't knew that :)

**Review by eneoli — APPROVED** [review]

**Comment by TornaxO7** [comment]
peepoAwesome

**Pull request merged**